Move Method::lowered_graph to a separate pass out of the Method class.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28242 Move Method::lowered_graph to a separate pass out of the Method class.**

There is no reason to have it in a general API of Module/Method - it's
just another graph pass. It was there because some time ago modules were
not first class and all graphs were lowered. After that changed, this
API was added for easier transition, but now we don't need it anymore.

Differential Revision: [D17986724](https://our.internmc.facebook.com/intern/diff/D17986724)

Commit:
Move Method::lowered_graph to a separate pass out of the Method class.

There is no reason to have it in a general API of Module/Method - it's
just another graph pass. It was there because some time ago modules were
not first class and all graphs were lowered. After that changed, this
API was added for easier transition, but now we don't need it anymore.

[ghstack-poisoned]

zdevito commented on torch/csrc/jit/passes/onnx.cpp:
```diff
@@ -355,5 +357,129 @@ void BlockToONNX(
+  return result;
+}
+
+std::pair<std::shared_ptr<Graph>, std::vector<at::Tensor>> ONNXLowerGraph(
```
Technically this is not ONNX specific so we probably should keep it as LowerGraph. It is probable that some backend integrations are using this function, and will need to be able to find it again after we move it.

zdevito reviewed (approved):
Thanks!

houseroad reviewed (approved):
LGTM

Commit:
Update on "Move Method::lowered_graph to a separate pass out of the Method class."


There is no reason to have it in a general API of Module/Method - it's
just another graph pass. It was there because some time ago modules were
not first class and all graphs were lowered. After that changed, this
API was added for easier transition, but now we don't need it anymore.

Differential Revision: [D17986724](https://our.internmc.facebook.com/intern/diff/D17986724)

[ghstack-poisoned]

Commit:
Update on "Move Method::lowered_graph to a separate pass out of the Method class."


There is no reason to have it in a general API of Module/Method - it's
just another graph pass. It was there because some time ago modules were
not first class and all graphs were lowered. After that changed, this
API was added for easier transition, but now we don't need it anymore.

Differential Revision: [D17986724](https://our.internmc.facebook.com/intern/diff/D17986724)

[ghstack-poisoned]
